internal/daemon/worker/worker.go

```diff
@@ -127,10 +127,11 @@ type Worker struct {
 
 	recorderManager recorderManager
 
-	everAuthenticated *ua.Uint32
-	lastStatusSuccess *atomic.Value
-	workerStartTime   time.Time
-	operationalState  *atomic.Value
+	everAuthenticated         *ua.Uint32
+	lastStatusSuccess         *atomic.Value
+	workerStartTime           time.Time
+	operationalState          *atomic.Value
+	controllerConnectionState *atomic.Value
```

Comment:
Should this be `UpstreamConnectionState` instead since some workers will be connected to an upstream worker?

Reply:
Updated to `upstreamConnectionState`